### PR TITLE
fix: correct trimIR out-of-bounds array access

### LINKED ISSUE #29: Energy Decay: trimIR out-of-bounds array access makes tail trimming non-functional
## Summary

The `trimIR` function has an off-by-one error that causes the tail-trimming logic to never execute, leaving trailing silence in impulse responses.

## Details

**File:** `src/compute/energy-decay.ts`, lines 267-271

```ts
function trimIR(ir: Float32Array): Float32Array {
    let tolerance = 1e-6; 
    let startSample = 0; 
    let endSample = ir.length; 

    // ... head trimming (lines 261-265) works correctly ...

    sindex = ir.length;                    // BUG: one past the last valid index
    while(Math.abs(ir[sindex]) < tolerance){  // ir[ir.length] is undefined
        endSample = sindex; 
        sindex--; 
    }

    return ir.slice(startSample, endSample); 
}
```

### Step-by-step failure

1. `sindex` is set to `ir.length` (e.g., 44100 for a 1-second IR at 44.1kHz)
2. `ir[44100]` accesses an out-of-bounds index — `Float32Array` returns `undefined`
3. `Math.abs(undefined)` returns `NaN`
4. `NaN < tolerance` evaluates to `false`
5. The while loop body **never executes**
6. `endSample` remains `ir.length` — no tail trimming occurs

### Impact

Impulse responses loaded into the Energy Decay solver retain all trailing near-zero samples. This:
- Wastes memory and computation in the octave-band filtering step
- May affect the Schroeder integration by including a long zero tail
- Could produce artificially long reverberation time estimates if the zero tail affects the -5 dB to -35 dB slope detection

## Expected behavior

The function should trim trailing samples below the tolerance threshold, returning only the meaningful portion of the impulse response.

## Suggested fix

Initialize `sindex` to the last valid index:

```ts
sindex = ir.length - 1;
while(sindex >= 0 && Math.abs(ir[sindex]) < tolerance){
    endSample = sindex; 
    sindex--; 
}
```

The added `sindex >= 0` guard prevents underflow if the entire array is below tolerance.

### PR DESCRIPTION
**Pull request by rdmiller**
## Summary
- `sindex` was initialized to `ir.length` (one past the last valid index)
- `ir[ir.length]` returns `undefined`, `Math.abs(undefined)` is `NaN`, `NaN < tolerance` is `false`
- The tail-trimming while loop never executed, leaving trailing silence untrimmed
- Fix: initialize `sindex = ir.length - 1`

## Test plan
- [x] Test proves buggy version fails to trim trailing silence
- [x] Test proves fixed version correctly trims trailing silence
- [x] Test demonstrates the NaN mechanism: `ir[ir.length]` → `undefined` → `NaN`
- [x] Tests verify: leading trim, trailing trim, both, content preservation, edge cases

Closes #29

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/compute/__tests__/trim-ir.spec.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  let endSample = ir.length;
+
+  let sindex = 0;
+  while (Math.abs(ir[sindex]) < tolerance) {
+    startSample = sindex;
+    sindex++;
+  }
+
+  sindex = ir.length - 1; // FIX: last valid index
+  while (Math.abs(ir[sindex]) < tolerance) {
```
[the remark this answers is not in the document]
Comment (rdmiller) [reply src/compute/__tests__/trim-ir.spec.ts]: Addressed — the extracted production function in trim-ir.ts has both bounds checks. The local duplicates in the test file are removed.

File: src/compute/__tests__/trim-ir.spec.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    it('handles single-sample impulse', () => {
+      const ir = new Float32Array([0, 0, 1.0, 0, 0]);
+      const result = trimIR_fixed(ir);
+
+      expect(result.length).toBeLessThanOrEqual(2);
+      expect(Math.max(...Array.from(result))).toBeCloseTo(1.0, 5);
+    });
+
+    it('respects the tolerance threshold', () => {
+      const tolerance = 1e-6;
```
[the remark this answers is not in the document]
Comment (rdmiller) [reply src/compute/__tests__/trim-ir.spec.ts]: Removed.
